### DISCUSSION TITLE
AggregationStateSerializerFactory is missing.

### DISCUSSION THREAD

**Discussion by jaehc** [trinodb/trino#23264] [Q&A]
Hi, 

I am currently migrating from presto to trino, specifically, from presto 0.211 to trino 358. 
One of the biggest parts of doing so is changing UDFs to be consistent with trino SPI.
Everything is fine, and most of the work has been done nicely, but except for one thing, which is related to this issue.

To begin with, I have several aggregation UDF functions, and some of them rely on parametric types, such that, 

```
public final class MyAggregation {

@InputFunction
@TypeParameter("K")
@TypeParameter("V")
 public static void input(
            @TypeParameter("K") Type keyElementType,
            @TypeParameter("V") Type valElementType,
            @AggregationState MyState state,
            @SqlType("map(K, array(V))") Block aBlock)
```
 
In the past, in order to get the instance of the custom AccumulatorStateSerializer, I had to use AggregationStateSerializerFactory annotation because it has to have some way to pass on parametric types, rather than being automatically generated by reflection. 
As seen in the code below, MyStateSerializer relies on parametric types and is provided with them through its constructor. 

```
public class MyStateSerializer
        implements AccumulatorStateSerializer<MyState>
{
    public MyStateSerializer(Type elementKeyType, Type elementValType)
    {
        this.elementKeyType = elementKeyType;
        this.elementValType = elementValType;
    }
```

The only way to call this constructor was through a function with AggregationStateSerializerFactory annotation that resides in the aggregation function class like the below.

```
public final class MyAggregation {

    ...Skip

    @AggregationStateSerializerFactory(MyState.class)
    public static MyStateSerializer createSerializer(
            @TypeParameter("K") Type keyElementType,
            @TypeParameter("V") Type valElementType)
    {
        return MyStateSerializer.create(keyElementType, new ArrayType(valElementType));
    }
```

However, AggregationStateSerializerFactory has been removed from the codebase as seen [here](https://github.com/trinodb/trino/commit/676daa84a1aea32e05862742dd2a37661ac0d651), and as of now, I don't have any particularly good idea of passing parametric types, that are K and V in this example, to the constructor. 

May I ask, 
Is there any good way to solve this problem or any other workaround?
Or, is it possible to reintroduce AggregationStateSerializerFactory to the codebase?

Thank you. :)

**Comment by hashhar** [comment]
Sorry that this fell through the gaps. Did you manage to find a solution?
Let us know if you still look for an answer for this and we'll try to find it for you.

Closing for now, will reopen upon hearing from you.
